### ISSUE TITLE
Figuring Out Async

### ISSUE THREAD

**Issue by wabbottRAL** [redarrowlabs/react-ts-boiler#7]
As we dug into the dispatch mapper we found it hard to follow the existing pattern and implement async actions.

**Comment by chriswxyz** [comment]
historically, we've used this pattern:

``` typescript
async function myAsyncAction(input: string){
    Store.dispatch({ type: MY_ACTION_PENDING });

    const result = await someAsyncCall();

    Store.dispatch({ type: MY_ACTION_SUCCESS, payload: result });
}
```

However, that does prevent you from using react-redux as-is. A more idiomatic redux implementation would probably use https://github.com/gaearon/redux-thunk or https://github.com/yelouafi/redux-saga

Will investigate...

**Comment by bovorasr** [comment]
I also saw redux-promise which looked promising for the promises-based swagger client

**Comment by chriswxyz** [comment]
Whoops, missed that comment. redux-thunk works with any Promise-based async.
